packages/blocks/src/surface-block/managers/edit-session.ts

```diff
@@ -0,0 +1,312 @@
+  StrokeStyle,
+} from '../consts.js';
+import type { EdgelessElementType } from '../edgeless-types.js';
+import {
+  ConnectorEndpointStyle,
+  DEFAULT_FRONT_END_POINT_STYLE,
+  DEFAULT_REAR_END_POINT_STYLE,
+} from '../elements/connector/types.js';
+import { TextAlign, TextVerticalAlign } from '../elements/consts.js';
+import {
+  DEFAULT_SHAPE_FILL_COLOR,
+  DEFAULT_SHAPE_STROKE_COLOR,
+  FillColorsSchema,
+  SHAPE_TEXT_FONT_SIZE,
+  ShapeType,
+  StrokeColorsSchema,
+} from '../elements/shape/consts.js';
+import type { SurfaceService } from '../surface-service.js';
+
+const ConnectorEndpointSchema = z.nativeEnum(ConnectorEndpointStyle);
+const StorkeStyleSchema = z.nativeEnum(StrokeStyle);
+const LineWidthSchema = z.nativeEnum(LineWidth);
+const ShapeStyleSchema = z.nativeEnum(ShapeStyle);
+const ShapeTextFontSizeSchema = z.nativeEnum(SHAPE_TEXT_FONT_SIZE);
+const CanvasTextFontFamilySchema = z.nativeEnum(CanvasTextFontFamily);
+const CanvasTextFontWeightSchema = z.nativeEnum(CanvasTextFontWeight);
+const CanvasTextFontStyleSchema = z.nativeEnum(CanvasTextFontStyle);
+const TextAlignSchema = z.nativeEnum(TextAlign);
+const TextVerticalAlignSchema = z.nativeEnum(TextVerticalAlign);
+const ShapeTypeSchema = z.nativeEnum(ShapeType);
+
+const LastPropsSchema = z.object({
+  connector: z.object({
+    frontEndpointStyle: ConnectorEndpointSchema,
+    rearEndpointStyle: ConnectorEndpointSchema,
+    strokeStyle: StorkeStyleSchema,
+    stroke: LineColorsSchema,
+    strokeWidth: LineWidthSchema,
+    rough: z.boolean(),
+    mode: z.number().optional(),
+  }),
+  brush: z.object({
+    color: LineColorsSchema,
+    lineWidth: LineWidthSchema,
+  }),
+  shape: z.object({
+    shapeType: ShapeTypeSchema,
+    fillColor: FillColorsSchema,
+    strokeColor: StrokeColorsSchema,
+    shapeStyle: ShapeStyleSchema,
+    filled: z.boolean(),
+    strokeWidth: z.number().optional(),
+    strokeStyle: StorkeStyleSchema.optional(),
+    color: z.string().optional(),
+    fontSize: ShapeTextFontSizeSchema.optional(),
+    fontFamily: CanvasTextFontFamilySchema.optional(),
+    fontWeight: CanvasTextFontWeightSchema.optional(),
+    fontStyle: CanvasTextFontStyleSchema.optional(),
+    textAlign: TextAlignSchema.optional(),
+    textHorizontalAlign: TextAlignSchema.optional(),
+    textVerticalAlign: TextVerticalAlignSchema.optional(),
+  }),
+  text: z.object({
+    color: z.string(),
+    fontFamily: CanvasTextFontFamilySchema,
+    textAlign: TextAlignSchema,
+    fontWeight: CanvasTextFontWeightSchema,
+    fontStyle: CanvasTextFontStyleSchema,
+    fontSize: z.number(),
+    hasMaxWidth: z.boolean(),
+  }),
+  'affine:note': z.object({
+    background: NoteColorsSchema,
+    hidden: z.boolean(),
+    edgeless: z.object({
+      style: z.object({
+        borderRadius: z.number(),
+        borderSize: z.number(),
+        borderStyle: StorkeStyleSchema,
+        shadowType: NoteShadowsSchema,
+      }),
+    }),
+  }),
+});
+
+const defaultLastProps = {
```

Comment:
we need to get the static keys of schema. So here we don't use .default().
.default() will add a wrapper causing not fetch the keys.